testing/xtrabackup/sigevent_musl.patch

```diff
@@ -0,0 +1,39 @@
+--- ./mysys/posix_timers.c.orig
++++ ./mysys/posix_timers.c
+@@ -22,6 +22,17 @@
+ #include <string.h>         /* memset */
+ #include <signal.h>
+
++struct linux_sigevent {
++    union sigval sigev_value;
++    int sigev_signo;
++    int sigev_notify;
++    union {
++       int _pad[64-sizeof(int) * 2 + sizeof(union sigval)];
++       int _tid;
++    } _sigev_un;
++};
++
++
+ #if defined(HAVE_SIGEV_THREAD_ID)
+ #include <sys/syscall.h>    /* SYS_gettid */
+
+@@ -191,7 +202,7 @@
+ int
+ my_timer_create(my_timer_t *timer)
+ {
+-  struct sigevent sigev;
++  struct linux_sigevent sigev;
+ 
+   memset(&sigev, 0, sizeof(sigev));
+ 
+@@ -200,7 +211,7 @@
+   sigev.sigev_notify= SIGEV_SIGNAL | SIGEV_THREAD_ID;
+   sigev.sigev_notify_thread_id= timer_notify_thread_id;
+ 
+-  return timer_create(CLOCK_MONOTONIC, &sigev, &timer->id);
++  return timer_create(CLOCK_MONOTONIC, (struct sigevent *)&sigev, &timer->id);
```

Comment:
Yeah, this will break horribly if you run the code with the above struct linux_sigevent definition.